lib/horde/dynamic_supervisor.ex

```diff
@@ -244,7 +244,8 @@ defmodule Horde.DynamicSupervisor do
 
   Works like `DynamicSupervisor.terminate_child/2`.
   """
-  @spec terminate_child(Supervisor.supervisor(), child_pid :: pid()) :: :ok | {:error, :not_found}
+  @spec terminate_child(Supervisor.supervisor(), child_pid :: pid()) ::
+          :ok | {:error, :not_found} | {:error, {:node_dead_or_shutting_down, String.t()}}
```

Comment:
OK so when I ran horde through dialyzer it failed on this because of the fact that terminate child returns `proxy_to_node` in some cases, and that introduces the possibility of terminate child returning `:node_dead_or_shutting_down` error. That's where this one comes from. 

Makes sense to me, what do you think?